backend/src/test/java/codezap/member/repository/MemberJpaRepositoryTest.java

```diff
@@ -1,45 +1,139 @@
+            Member member = MemberFixture.getFirstMember();
+            member = memberRepository.save(member);
 
-        assertThat(memberJpaRepository.fetchByTemplateId(savedTemplate.getId())).isEqualTo(member);
+            Member actual = memberRepository.fetchById(member.getId());
+
+            assertThat(actual).isEqualTo(member);
+        }
+
+        @Test
+        @DisplayName("실패 : 존재하지 않는 id인 경우 에러가 발생한다.")
+        void fetchByIdFailByNotExistsId() {
+            long id = 100;
+
+            assertThatThrownBy(() -> memberRepository.fetchById(id))
+                    .isInstanceOf(CodeZapException.class)
+                    .hasMessage("식별자 " + id + "에 해당하는 멤버가 존재하지 않습니다.");
+        }
+    }
+
+    @Nested
+    @DisplayName("아이디로 Member 조회")
+    class fetchByName {
+        @Test
+        @DisplayName("성공 : 아이디로 Member를 알아낼 수 있다.")
+        void fetchByNameSuccess() {
+            Member member = MemberFixture.getFirstMember();
+            member = memberRepository.save(member);
+
+            Member actual = memberRepository.fetchByName(member.getName());
+
+            assertThat(actual).isEqualTo(member);
+        }
+
+        @Test
+        @DisplayName("실패 : 존재하지 않는 아이디인 경우 에러가 발생한다.")
+        void fetchByNameFailByNotExistsId() {
+            String name = "켬미";
+
+            assertThatThrownBy(() -> memberRepository.fetchByName(name))
+                    .isInstanceOf(CodeZapException.class)
+                    .hasMessage("존재하지 않는 아이디 " + name + " 입니다.");
+        }
+    }
+
+    @Nested
+    @DisplayName("템플릿 id로 Member 조회")
+    class fetchByTemplateId {
+        @Test
+        @DisplayName("성공 : 템플릿 id로 Member를 알아낼 수 있다.")
+        void fetchByTemplateIdSuccess() {
+            Member member = memberRepository.save(MemberFixture.getFirstMember());
+            Category category = categoryRepository.save(CategoryFixture.getFirstCategory());
+            Template template = new Template(member, "title", "description", category);
+            template = templateRepository.save(template);
+
+            Member actual = memberRepository.fetchByTemplateId(template.getId());
+
+            assertThat(actual).isEqualTo(member);
+        }
+
+        @Test
+        @DisplayName("실패 : 존재하지 않는 템플릿 id인 경우 에러가 발생한다.")
+        void fetchByTemplateIdFailByNotExistsId() {
+            long id = 100;
+
+            assertThatThrownBy(() -> memberRepository.fetchByTemplateId(id))
+                    .isInstanceOf(CodeZapException.class)
+                    .hasMessage("템플릿에 대한 멤버가 존재하지 않습니다.");
+        }
+    }
+
+    @Nested
+    @DisplayName("아이디가 존재 여부")
+    class existsByName {
+        @Test
+        @DisplayName("성공 : 해당 아이디가 존재하면 true를 반환한다.")
+        void existsByNameReturnTrue() {
+            Member member = memberRepository.save(MemberFixture.getFirstMember());
+
+            boolean actual = memberRepository.existsByName(member.getName());
+
+            assertThat(actual).isTrue();
```

Comment:
```suggestion
            assertThat(memberRepository.existsByName(member.getName()))
                    .isTrue();
```

[제안]
확인하는 내용을 한 눈에 볼 수 있도록 변수 추출을 제거해 보았는데요. 
가볍게 확인 해 보시고 켬미가 보기에 더 편한 것으로 선택해 주세요!

Reply:
좋은 의견 같아요 ~ 하지만 저는 when then이 분리되어있는게 더 좋아보여서 고치지 않겠습니다 ~!